### PR TITLE
Add temporal database support and extended test coverage

### PR DESCRIPTION
**Pull request by tolbrino**
In addition this pull request introduces several documentation, configuration, and dependency improvements to the project, focusing on clearer development guidelines, enhanced architecture documentation, and improved workspace organization. 

## Summary by CodeRabbit

* **New Features**
  * GraphQL transaction mutations (fire‑and‑forget, async, sync) with persisted transaction records; RPC‑backed balance & allowance queries; historical snapshot + live opened‑channels subscription; standalone API mode.

* **Improvements**
  * Background transaction monitor, transaction status tracking, reorg detection & corrective state handling; richer GraphQL union results and validation; schema wired to indexer/RPC state for end‑to‑end flows.

* **Documentation**
  * Expanded developer, architecture, testing, and quickstart guides.

* **Tests**
  * Extensive new unit and integration test suites.

* **Chores**
  * Project metadata, ignore rules, attributes, and issue/metadata additions.

### CODE REVIEW

File: api/src/lib.rs [hunk 1]
```diff
@@ -66,8 +80,65 @@ pub async fn start_server(network: String, config: ApiConfig) -> ApiResult<()> {
...
+    // are typically called through bloklid, not standalone API
+    warn!("Running in standalone mode - transaction mutations will not work without bloklid");
+
+    let transaction_store = Arc::new(TransactionStore::new());
+    let transaction_validator = Arc::new(TransactionValidator::new());
+
+    // Create RPC connection for balance queries
+    info!("Connecting to RPC: {}", config.rpc_url);
+    let transport_client = alloy::transports::http::ReqwestTransport::new(
+        url::Url::parse(&config.rpc_url).expect("Failed to parse RPC URL"),
```
[the remark this answers is not in the document]
Comment (tolbrino) [reply api/src/lib.rs]: @coderabbitai This has been fixed.

File: api/tests/balance_api_integration_test.rs [hunk 2]
```diff
@@ -0,0 +1,576 @@
...
+        if let Some(balance_str) = balance {
+            let _parsed =
+                HoprBalance::from_str(&balance_str).expect("Balance string should be valid HoprBalance format");
+        }
+
+        assert!(
+            duration < Duration::from_millis(10),
+            "Balance query took too long: {:?}",
+            duration
+        );
```
[the remark this answers is not in the document]
Comment (tolbrino) [reply api/tests/balance_api_integration_test.rs]: the limit is ok for now

File: TESTING.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+- [Running Tests](#running-tests)
+- [Test Organization](#test-organization)
+- [Core Testing Areas](#core-testing-areas)
+- [Temporal Query Testing](#temporal-query-testing)
+- [Reorg Testing](#reorg-testing)
+- [Edge Case Testing](#edge-case-testing)
+- [Performance Testing](#performance-testing)
+- [Writing New Tests](#writing-new-tests)
+
```
[the remark this answers is not in the document]
Comment (tolbrino) [reply TESTING.md]: Has been fixed.

File: chain/api/src/transaction_monitor.rs [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+                    Err(e) => {
+                        error!("Error checking transaction {}: {}", record.id, e);
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
```
[the remark this answers is not in the document]
Comment (tolbrino) [reply chain/api/src/transaction_monitor.rs]: Improved.
